### ISSUE TITLE
Returning Dependent Pack Expansions

### ISSUE THREAD

**Issue by ricejasonf** [ricejasonf/parametric_expressions#7]
Currently the compiler eagerly fails if a dependent expression tries to expand a parameter pack if it can't detect an unexpanded pack.

Is it going too far to defer this check for dependent expressions?

https://godbolt.org/z/S5lxNj

```cpp
template <auto ...i>
struct iota_
{
    static using apply() {
        return i;
    }
};
  
template <auto N>
struct test {
    void run() {
        (iota_<N>::apply(), ...);
    }
};
```
```
<source>:12:29: error: pack expansion does not contain any unexpanded parameter packs

        (iota_<N>::apply(), ...);

         ~~~~~~~~~~~~~~~~~  ^
```

**Comment by ricejasonf** [comment]
A parmexpr could never return a pack in one case but not another so there is no value in resolving that late. It was discussed on Slack, and there is also a case of ambiguity with packs of packs or base expression packs yielding packs.

There is actually a blog post about it:
https://quuxplusone.github.io/blog/2019/02/11/tilde-notation-for-exploding-tuples/

To fix the dependence issue we can add a declaration syntax and call-site marker `~` so Sema knows to expect a pack no matter what.

Something like:
```cpp
template <auto ...i>
struct iota_
{
    static using apply~() {
        return i;
            // ^ must contain unexpanded pack
    }
};

(iota_<Ns...>::apply~() + ...);
```
This means that we can enforce that the parmexpr must return a pack.

To fix the ambiguity, we can simply enforce that the input arguments can not be unexpanded parameter packs. With an instance member expression this is easy because it is the first argument. If it is too difficult or impossible to do with qualified lookups then we can disallow them to be declared `static`. Operator overloading for these won't be supported either.

**Comment by ricejasonf** [comment]
Fixed in 2033a1c2fcea365bff23b0a2208ab931c655b353 in `f_alias` branch.

Also added is the `operator()~` postfix tilde operator to allow converting objects into packs like
```cpp
my_array xs = {0, 1, 2, 3 4};
int sum = (xs~ + ...);
```
This involves duplicate evaluation in the operand of `~`, but I think that is intuitive since it appears in a pack expansion.
